connection-loader/src/main/java/com/terracotta/connection/api/AbstractConnectionService.java

```diff
@@ -48,6 +50,23 @@ public final Connection connect(URI uri, Properties properties) throws Connectio
     return internalConnect(clientConfig);
   }
 
+  private void validateURI(URI uri) throws ConnectionException {
```

Comment:
Can be static and in fact could be in some util class. I would love to see unit tests on that.

Reply:
I can do that.